src/main/scala/dpla/ingestion3/utils/OutputHelper.scala

```diff
@@ -1,38 +1,173 @@
+   */
+  lazy val outputPath: String = s"$directory$fileKey"
+
+  /*
+   * Parse S3 bucket name from given `root'.
+   * Does not include trailing slash or "s3a://" prefix.
+   * Returns empty string if unable to parse bucket name.
+   */
+  lazy val bucketName: String = Try{ directory.split("/")(2) }.getOrElse("")
+
+  /*
+   * Get path to manifest file, not including local root directory or s3 bucket.
+   * Manifest will be in the same directory as activity output files.
+   * Does not include starting "/".
+   */
+  lazy val manifestKey: String = s"$fileKey/_MANIFEST"
+
+  /*
+   * Get path to manifest with local root directory.
+   */
+  lazy val manifestLocalOutPath: String = s"$directory$manifestKey"
+
+  /*
+   * Get directory for temporary files.
+   * If system property is not set, uses "/tmp" as default.
+   * TODO: Move to more general File IO Util
+   */
+  lazy val tempDirectory: String = System.getProperty("java.io.tmpdir", "/tmp")
+
+  // TODO Move to more general File IO Util
+  lazy val s3client: AmazonS3Client = new AmazonS3Client()
+
+  /*
+   * Write a manifest file in the given outputPath directory.
+   *
+   * @param outputPath: The directory in which the manifest file is to be written.
+   * @param opts: Optional data points to be included in the manifest file.
+   */
+  def writeManifest(opts: Map[String, String]): Unit = {
+
+    val text: String = manifestText(opts)
+
+    if (outputPath.startsWith("s3a://"))
+      writeS3File(bucketName, manifestKey, text)
+    else
+      writeLocalFile(manifestLocalOutPath, text)
+  }
+
+  /*
+   * Create text for a manifest file.
+   *
+   * @param opts: Optional data points to be included in the manifest file.
+   *              This is intentionally open-ended so that individual executors
+   *              can include whatever data points are relevant to their activity.
+   */
+  val manifestText: Map[String, String] => String = (opts: Map[String, String]) => {
+
+    val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+
+    // Add date/time to given `opts'
+    val data: Map[String, String] =
+      opts + ("Start date/time" -> startDateTime.format(formatter))
+
+    data.map{ case(k, v) => s"$k: $v" }.mkString("\n")
+  }
+
+  // Write a String to a local file.
+  // TODO: Move to more general FileIO Util
+  def writeLocalFile(outPath: String, text: String): File = {
+    val file: File = new File(outPath)
+    file.getParentFile.mkdirs
+    val bw: BufferedWriter = new BufferedWriter(new FileWriter(file))
+    bw.write(text)
+    bw.close()
+    file
+  }
+
+  // Write a String to an S3 file.
+  // TODO: Move to a more general FileIO Util
+  def writeS3File(bucket: String, key: String, text: String): PutObjectResult = {
+    // Write to a temporary local file
+    val tempOut: String = s"$tempDirectory/$key"
+    val tempFile: File = writeLocalFile(tempOut, text)
```

Comment:
`putObject` has an alternative implementation that takes an InputStream, so you don't need to save locally first. It would go something like:
```
val in = new ByteArrayInputStream(text.getBytes("utf-8"))
s3client.putObject(new PutObjectRequest(bucket, key, in, new ObjectMetadata())
```

Reply:
@mdellabitta I saw that but I was did not understand what `ObjectMetadata` was all about.  Have you used it before?

Reply:
It's sort of like HTTP headers... you can set a mime type, etc. Not especially relevant.